src/algebra/ordered_sub.lean

```diff
@@ -0,0 +1,680 @@
+/-
+Copyright (c) 2021 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import algebra.ordered_monoid
+/-!
+# Ordered Subtraction
+
+This file proves lemmas relating (truncated) subtraction with an order. We provide a class
+ `has_ordered_sub` stating that `a - b ≤ c ↔ a ≤ c + b`.
+
+The subtraction discussed here could both be normal subtraction in an additive group or truncated
+subtraction on a canonically ordered monoid (`ℕ`, `multiset`, `enat`, `ennreal`, ...)
```

Comment:
AFAICT, you only provide a `has_ordered_sub ℕ` instance. Do you have plans to provide instances for the other types listed here (in follow-up PRs)?

Reply:
I guess `finsupp` is also an example? To be precise `X ->\_0 A`, whenever `A` has ordered sub. I'm guessing this is the abstraction that I've needed before a couple of times.
I'm just going to leave this comment, so that hopefully we think about this instance in one of those future PRs :stuck_out_tongue_winking_eye: